### PR TITLE
LPD-17755 Add Cards visualization mode

### PR DESCRIPTION
**Pull request by p2kmgcl**
This is basically a copy of List visualization mode with different fields. I thought it was a good idea to just duplicate the code instead of creating an extra abstraction because:
1. There is not much of it.
2. We don't know how much are they going to differ in a near future.

### CODE REVIEW

**Comment by markocikos** [comment]
@p2kmgcl  Please see https://github.com/liferay-frontend/liferay-portal/pull/3990, we are significantly changing structures and naming in lists, which will likely affect cards as well.

**Comment by p2kmgcl** [comment]
> @p2kmgcl Please see https://github.com/liferay-frontend/liferay-portal/pull/3990, we are significantly changing structures and naming in lists, which will likely affect cards as well.

Sure! I'll wait until that is forwarded to apply any extra changes. I am also going to send a PR adjusting some styles that I checked with Uge this morning and need to be applied here.

**Comment by p2kmgcl** [comment]
@markocikos I think this is ready to be reviewed, but the backend does not work. I added a new method to the FDSViewPortlet but I am sure I am missing more things.

Anyway, the frontend works fine (it does not crash or anything else).

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java [hunk 1]
```diff
@@ -471,6 +471,47 @@ private void _createFDSFieldObjectDefinition(
 			ObjectRelationshipConstants.TYPE_ONE_TO_MANY, null);
 	}
 
+	private void _createFDSCardsSectionObjectDefinition(
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:474]: This code is technically perfectly fine, but Brian would not allow a private method that is unused. You can remove this, and I'll wire backend in a followup, or you can soldier through and add missing bits:

1. This needs to be [invoked](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java#L672-L675), behind FF, along with `FDSListSection`.
1. New [modifiable system object](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-api/src/main/java/com/liferay/object/definition/util/ObjectDefinitionUtil.java#L122). This also defines a new endpoint alias, `/data-set-manager/cards-sections`
1. New endpoint in [constants](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/utils/constants.ts#L14) , and using it as `API_URL.FDS_CARDS_SECTIONS`

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -19383,6 +19384,7 @@ this-value-cannot-be-less-than-1=This value cannot be less than 1.
 this-value-is-used-for-alt-text=This value is used as ALT text.
 this-version=This version
 this-version-will-be-published-when-x-is-published-to-live=This version will be published when <em>{0}</em> is published to live.
+this-visualization-mode-will-not-be-shown-until-you-assign-at-least-one-field-to-a-card-element=This visualization mode won't be shown until you assign at least one field to a card element.
```
Comment (markocikos) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:19387]: [nit] `won't` should probably be `will not` in both values.

**Review by markocikos — COMMENTED** [review]
@p2kmgcl please see comments inline

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java:474]: I'm gonna try to finish everything :smile: 

Thanks!

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 2 above)
Comment (p2kmgcl) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:19387]: Right!

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java]: It works now :heart_eyes:

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 2 above)
Comment (markocikos) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:19387]: I'm not sure either. Not blocking for PR, in any case.

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by p2kmgcl** [comment]
ci:forward:force

**Pull request closed without merging**